core/shared/src/main/scala/scalaz/zio/Queue.scala

```diff
@@ -17,7 +17,7 @@ import Queue.internal._
  * 2. Benchmark to see how slow this implementation is and if there are any
  *    easy ways to improve performance.
  */
-class Queue[A] private (capacity: Int, ref: Ref[State[A]]) {
+class Queue[A] private (capacity: Int, ref: Ref[State[A]]) extends Serializable {
```

Comment:
In order for this to have any meaning, the `State` and of course `Ref` must be serializable.

Reply:
We can change `Ref` so it is no longer `AnyVal`.